src/NuGet.Core/NuGet.Protocol/PublicAPI/netstandard2.0/PublicAPI.Unshipped.txt

```diff
@@ -11,3 +11,4 @@ NuGet.Protocol.PackageVulnerabilityMetadata.Severity.get -> int
 const NuGet.Protocol.JsonProperties.AdvisoryUrl = "advisoryUrl" -> string
 const NuGet.Protocol.JsonProperties.Severity = "severity" -> string
 const NuGet.Protocol.JsonProperties.Vulnerabilities = "vulnerabilities" -> string
+NuGet.Protocol.Core.Types.PackageUpdateResource.Push(System.Collections.Generic.IList<string> packagePaths, string symbolSource, int timeoutInSecond, bool disableBuffering, System.Func<string, string> getApiKey, System.Func<string, string> getSymbolApiKey, bool noServiceEndpoint, bool skipDuplicate, NuGet.Protocol.Core.Types.SymbolPackageUpdateResourceV3 symbolPackageUpdateResource, NuGet.Common.ILogger log) -> System.Threading.Tasks.Task
```

Comment:
From an API design point of view, I'm not really a fan. I'd prefer if whatever calls `NuGet.Protocol` was responsible for iterating it.  Any ideas how the `*.nupkg` wildcard is expanded? Is it done in `NuGet.Commands`, or `NuGet.Protocol`?

Reply:
The `*.nupkg` is either expanded by the shell or it is expanded in `NuGet.Protocol` (when it is not expanded by the shell) by this call: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs#L217

I can move the loop to the `NuGet.Commands` which calls the `Push` method from `NuGet.Protocol`. This way the API of `NuGet.Protocol` will remain unmodified. Should I do it ?

Reply:
nah, I didn't know (or maybe just forgot) that NuGet.Protocol does the file globbing 👎 . Since it's already possible for `PackageUpdateResource` to push multiple files with one invocation, your change is very reasonable.

I'll just continue to wish our public APIs were better designed, but I'd say you're done the right thing.